scripts/cmd/gocodegen/gocodegen.go

```diff
@@ -286,7 +286,7 @@ func goDataType(fieldName, elasticsearchDataType string) string {
 		return "time.Time"
 	case "boolean":
 		return "bool"
-	case "object":
+	case "object", "nested":
```

Comment:
@andrewkroh Foresee any issues with adding this mapping for `nested`?

Reply:
For `nested` maybe we should generate a more strict data type (one that incorporates the nested object's schema). Assuming we know the schema for the nested object then generating a type like `[]Segment` (in the case of PE segments) would be superior IMO.

That would look like:

```
type Pe struct {
    Segments []Segment
}

type Segment {
    Type string
    Sections string
}

or 

type Pe struct {
    Segments []struct{
        Type string
        Sections string
    }
}
```

Reply:
Yeah, I understand how having the schema more strictly defined would be superior. Unfortunately, this is currently a blocker to adding `nested` and `flattened` types since not having the top defined here causes an exception when generating a new set of artifacts. 

If this is something that we definitely should do right, I'm more than happy to help make that happen. Though, do we know if anything consuming the code generated here relies on `nested`? I prefer to avoid sinking time into a larger effort if the need isn't really there right now.

Reply:
@ebeahan to my knowledge the only thing on the Elastic side of things that consumes this right now is Packetbeat, which doesn't use any nested fields.